### PR TITLE
adding splash activity and main activity UI

### PR DESCRIPTION
**Pull request by benjaglobant**
In this PR I have implemented:
 - SplashScreen using motionlayout, viewmodel, coroutines and viewbinding.
 - MainActivity UI (In the next PR I will implement this activity)
 - Test for the viewmodel used in SplashActivity
SplashScreen + MainActivity gif (normal device):
splashPRJ7
SplashScreen + MainActivity gif (small device):
splashPRNexusS
Test image:
image
Coverage image:
image

### CODE REVIEW

File: app/src/main/java/com/globant/harrypotterapp/viewmodel/SplashViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    override fun startTimer() = viewModelScope.launch {
+        delay(TIME)
+        splashStateMutableLiveData.postValue(SplashState.START)
+        delay(TIME)
+        splashStateMutableLiveData.postValue(SplashState.FINISH)
+    }
+
+    companion object {
+        const val TIME = 2000L
```
Comment (lucianoeli) [inline app/src/main/java/com/globant/harrypotterapp/viewmodel/SplashViewModel.kt]: use a more representative name please

File: app/src/main/java/com/globant/harrypotterapp/activity/SplashActivity.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        setContentView(binding.root)
+        splashViewModel.getSplashStateLiveData().observe(::getLifecycle, ::runAnimation)
+        splashViewModel.startTimer()
+    }
+
+    private fun runAnimation(state: SplashState) {
+        when (state) {
+            SplashState.START -> binding.logo.performClick()
+            SplashState.FINISH -> {
+                startActivity(Intent(this@SplashActivity, MainActivity::class.java))
```
Comment (mauroLonghi) [inline app/src/main/java/com/globant/harrypotterapp/activity/SplashActivity.kt]: Pls add into the componion obj the intent

**Review by SartiMau — APPROVED** [review]
All the .png images can't be moved to .webp?

File: app/src/main/java/com/globant/harrypotterapp/viewmodel/SplashViewModel.kt (hunk 1 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/harrypotterapp/viewmodel/SplashViewModel.kt]: Refactored!

File: app/src/main/java/com/globant/harrypotterapp/activity/SplashActivity.kt (hunk 2 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/harrypotterapp/activity/SplashActivity.kt]: Done!

File: app/src/main/java/com/globant/harrypotterapp/activity/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+class MainActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+    }
+    // TODO: implement SpellActivity
+
+    companion object {
+        fun getIntent(activity: Activity): Intent {
```
Comment (SartiMau) [inline app/src/main/java/com/globant/harrypotterapp/activity/MainActivity.kt]: you can use this here:
fun getIntent(activity: Activity): Intent = Intent(activity, MainActivity::class.java)

Comment (benjaglobant) [reply app/src/main/java/com/globant/harrypotterapp/activity/MainActivity.kt]: Yes, you are right, thanks Mau! Refactored

**Review by SartiMau — APPROVED** [review]

File: app/src/main/res/layout/activity_splash.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    <ImageView
+        android:id="@+id/logo"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        android:layout_margin="@dimen/splash_activity_hogwarts_image_view_margin"
+        android:src="@drawable/hogwarts"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
```
Comment (nicolbidone) [inline app/src/main/res/layout/activity_splash.xml]: you can omit these constraints by using just "match_parent" in height and width

**Review by SartiMau — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

File: app/src/main/res/layout/activity_splash.xml (hunk 4 above)
Comment (benjaglobant) [reply app/src/main/res/layout/activity_splash.xml]: Yes! Refactored

**Review by SartiMau — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Review by hernanhorquin96 — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Pull request merged**